### ISSUE TITLE
export more useful functions from Chronos, and other suggestions

### ISSUE THREAD

**Issue by chessai** [andrewthad/chronos#12]
Many of the following functions I think are generally useful, while others are potentially useful to users, but none are exported.

```Haskell
monthLength,
isLeapYear,
ordinalDateToDay,
monthDateToDayOfYear,
dayOfYearToMonthAndDay,
dayToOrdinalDate,
parseSecondsAndNanoseconds,
countZeroes,
nanosecondsBuilder, 
microsecondsBuilder,
millisecondsBuilder,
prettyNanosecondsBuilder,
parseSecondsAndNanosecondsUtf8,
countZeroesUtf8,
nanosecondsBuilderUtf8,
microsecondsBuilderUtf8,
millisecondsBuilderUtf8,
prettyNanosecondsBuilderUtf8,
int64Builder, -- Does this even need to be defined here?
-- ** Not sure about offset stuff, but it seems like it could be useful
offsetBuilder, 
offsetParser,
parseSignedness,
parseOffset_z,
parseOffset_z1,
parseOffset_z2,
parseOffset_z3,
buildOffset_z,
buildOffset_z1,
buildOffset_z2,
buildOffset_z3,
parseSignednessUtf8,
parseOffsetUtf8_z,
parseOffsetUtf8_z1,
parseOffsetUtf8_z2,
parseOffsetUtf8_z3,
buildOffsetUtf8_z,
buildOffsetUtf8_z1,
buildOffsetUtf8_z2,
buildOffsetUtf8_z3,
-- ** End of offset stuff
--
-- ** Not sure about this stuff
countDigits,
clip,
parseFixedDigits,
parseFixedDigitsIntBS,
indexTwoDigitTextBuilder,
indexTwoDigitByteStringBuilder,
twoDigitByteStringBuilder,
twoDigitTextBuilder,
monthToZeroPaddedDigit,
zeroPadDayOfMonth,
monthToZeroPaddedDigitBS,
zeroPadDayOfMonthBS
```

Additionally, the following are a list of all official UTC Offsets, as integers:

100,200,300,330,400,430,500,530,545,600,630,700,800,845,900,930,1000,1030,1100,1200,1245,1300,1345,1400,0,-100,-200,-230,-300,-330,-400,-500,-600,-700,-800,-900,-930,-1000,-1100,-1200

Lastly, I think it would be useful to provide Enum and Bounded instances for things that do not currently have them (e.g. Offset), however, this could prove to be less user friendly by taking away some flexibility that users have.

EDIT: Also providing Torsor instances for things such as Offset.

**Comment by andrewthad** [comment]
In b91d9d27e70b7a4732d82c0599a73724f52b0098, I've added the following functions. Some of these were actually things that I had intended to export originally but had accidentally left out.

```
monthDateToDayOfYear
dayOfYearToMonthDay
daysInMonth
isLeapYear
observedOffsets
builderOffset
parserOffset
builderOffsetUtf8
parserOffsetUtf8
```

I've also added a `Torsor` and `Enum` instance for `Offset` but not a `Bounded` instance, since the only `Bounded` instance I'd really be comfortable with would be the one that provides the lowest and highest machine integers, which is basically never what anyone would want. Also, see https://github.com/andrewthad/chronos/issues/13

The other functions, while possibly generally useful, are things that I'd rather not have as part of the public API. In the event that someone has a particular need for one of them, I'm definitely open to discuss it, but without an use case in mind, I'm not convinced that they would actually ever be generally useful in their current forms. (Although, I could definitely see a `builderTimespan` and `parserTimespan` function that builds on top on things like `prettyNanosecondsBuilder` being useful, but even in that case, `prettyNanosecondsBuilder` still wouldn't need to get exported).

Check over what I've added and close out this issue unless I messed something up or unless there's one of the functions you wanted to make a stronger case for.
